base/shared/src/main/scala/scalaz/data/ilist.scala

```diff
@@ -37,6 +41,87 @@ trait IListModule {
   implicit def isCovariantInstance: IsCovariant[IList]
 }
 
+trait IListSyntax {
+  final class ToIListOps[A](self: IList[A]) {
+
+    def headOption: Maybe[A] =
+      IList.uncons(self) match {
+        case Maybe2.Empty2()    => Maybe.empty
+        case Maybe2.Just2(a, _) => Maybe.just(a)
+      }
+
+    def tailOption: Maybe[IList[A]] =
+      IList.uncons(self) match {
+        case Maybe2.Empty2()     => Maybe.empty
+        case Maybe2.Just2(_, as) => Maybe.just(as)
+      }
```

Comment:
Could we call these `head` and `tail`?

Reply:
sure - i'm on board with making the safe versions mandatory 😄